### PR TITLE
Support image screenshot for drivers other than Selenium2Driver

### PR DESCRIPTION
**Pull request by piggito**
Fixes issue #13 

`getScreenshot()` is defined in https://github.com/minkphp/Mink/blob/master/src/Driver/DriverInterface.php#L246 so we shouldn't limit image screenshot to `Selenium2Driver`

I used `getimagesizefromstring` to detect the image type and set the correct file extension.

### CODE REVIEW

**Comment by piggito** [comment]
Added image file extensions other than PNG to autoClean

**Comment by forceedge01** [comment]
@piggito thanks for raising this PR. Firstly, although the screenshot method is defined in the interface, some drivers simply throw an exception as they don't have a concept of an actual screenshot. Drivers that rely on guzzle for example would all do this. These are only supported by drivers that have javascript capability which is why it has that certain condition on it. I will at alternative ways to achieve this but thanks for this.

The image extensions, not sure if that would be useful? The reason why I had one type is to lessen the complications around image generation and other manipulations. Have you experienced an issue with the png images?

File: bootstrap/Context/FailureContext.php [hunk 1]
```diff
@@ -413,12 +412,13 @@ public function takeScreenshot($filename, ElementInterface $page, $driver)
         $content = null;
         $filename .= microtime(true);
         // If not selenium driver, extract the html and put in file.
-        if ($this->screenshotMode === self::SCREENSHOT_MODE_HTML || ! ($driver instanceof Selenium2Driver)) {
+        if ($this->screenshotMode === self::SCREENSHOT_MODE_HTML || ! ($driver instanceof DriverInterface)) {
```
Comment (forceedge01) [inline bootstrap/Context/FailureContext.php]: We can probably do it this way whilst adding a try catch method on the getScreenshot call for other drivers that don't actually support the method and fallback onto the html screenshot? Also this extension is covered by tdd and bdd (for bdd - not passing test but a visual test more so). Would be good to get those in as well.

File: bootstrap/Context/FailureContext.php [hunk 2]
```diff
@@ -413,12 +412,13 @@ public function takeScreenshot($filename, ElementInterface $page, $driver)
...
             $filename .= '.html';
             $content = $this->applySiteSpecificFilters($page->getOuterHtml());
         } else {
-            $filename .= '.png';
             $content = $driver->getScreenshot();
+            $size = getimagesizefromstring($content);
```
Comment (forceedge01) [inline bootstrap/Context/FailureContext.php]: As commented previously, don't believe this is needed as png's have been consistent and easier to predict the behaviour and tools that can integrate with this extension further. If anything, this could be raised as another PR and we can have a chat about its usefulness thr. I do thank you for your contribution as this PR alone makes things easier :)

Comment (piggito) [reply bootstrap/Context/FailureContext.php]: I included this change cause `Behat\Mink\Driver\DriverInterface` doesn't constraint to a specific image format. I agree though we could have another PR.

**Comment by forceedge01** [comment]
Hello, thanks for your PR - i've resolved this issue in a slightly different way. I've got packs that use the default mode to take screenshots as png and html where supported - for this reason i've created the following PR https://github.com/forceedge01/behat-fail-aid/pull/20, hope this resolves your issue too.

**Comment by forceedge01** [comment]
This issue is now resolved in a slightly different way.
